app/main.py

```diff
@@ -1 +1,56 @@
-#  write your code here
+from __future__ import annotations
+
+
+class Animal:
+    def __init__(
+            self,
+            name: str,
+            appetite: int,
+            is_hungry: bool = True
+    ) -> None:
+        self.name = name
+        self.appetite = appetite
+        self.is_hungry = is_hungry
+
+    def print_name(self) -> None:
+        print(f"Hello, I'm {self.name}")
+
+    def feed(self) -> int:
+        if self.is_hungry:
+            print(f"Eating {self.appetite} food points...")
+            food_points = self.appetite
+            self.is_hungry = False
+            self.appetite = 0
+            return food_points
+        else:
+            self.appetite = 0
+            return self.appetite
+
+
+class Cat(Animal):
+    def __init__(self, name: str, is_hungry: bool = True) -> None:
+        super().__init__(name, appetite=3)
+        self.name = name
+        self.is_hungry = is_hungry
```

Comment:
No need to do like this. You can pass all parameters with to `super().__init__()`

Reply:
The same in dog class

Reply:
https://realpython.com/python-super/#super-in-single-inheritance